packages/aws-rfdk/lib/core/scripts/bash/mountEfs.sh

```diff
@@ -58,6 +63,68 @@ function use_nfs_mount() {
   return $?
 }
 
+function resolve_mount_target_ip_via_api() {
+  local EFS_FS_ID=$1
+  local MNT_TARGET_RESOURCE_ID=$2
+  local AVAILABILITY_ZONE_NAME=$3
+  local AWS_REGION=$4
+  local MOUNT_POINT_IP=""
+
+  if [[ $MNT_TARGET_RESOURCE_ID == fs-* ]]; then
+    # Mounting without an access point
+    MOUNT_POINT_IP=$(aws efs describe-mount-targets \
```

Comment:
Missing failure handling -- we're not guaranteed that there is a MountTarget in the same AZ as the instance. Route53 would resolve the DNS with a randomly chosen IP in that case.

Same in the `fsap-*` case

Reply:
Given the concern you mentioned about cross-AZ data charges, do you think we should replicate this behavior? Or is it better to fail? I'm learning towards failing with a user-friendly warning over using a random mount target. That way anyone using this could avoid the cross-AZ costs and simply add the missing mount target.

The calling code has a catch-all error handler, but you're right - currently this will not fail gracefully if there's no matching mount target. I'll fix that - thanks!

Reply:
My opinion... We want to make sure that the customer knows that there could be cross-AZ data charges, but not prevent them from deploying in a cross-AZ manner. Throwing an error for cross-AZ would make a cross-AZ mount impossible, and there may be valid use-cases for that.

Reply:
I've added logic to fall-back to using any available mount target with a warning being output that should get logged to CloudWatch. Let me know what you think.